source/common/quic/envoy_quic_client_session.cc

```diff
@@ -208,5 +216,12 @@ void EnvoyQuicClientSession::OnNewEncryptionKeyAvailable(
   }
 }
 
+void EnvoyQuicClientSession::ProcessUdpPacket(const quic::QuicSocketAddress& self_address,
+                                              const quic::QuicSocketAddress& peer_address,
+                                              const quic::QuicReceivedPacket& packet) {
+  ScopeTrackerScopeState scope(this, dispatcher());
```

Comment:
I'm not 100% up to speed on ScopeTrackerScopeState. Does this help make dumpState() work better, or is this an orthogonal improvement?

Reply:
It's an RAII object registering to the dispatcher that we're working on this ClientSession. When we crash, we'll go through all these registered objects and invoke `dumpState()` on them

Reply:
Ah! That makes perfect sense. Thanks!